### PR TITLE
Add standard moving targets that the ball can hit and will disappear after hit

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by jonathanraes** [comment]
May be best to merge this to COmbo and then Combo to master

**Comment by jonathanraes** [comment]
Combo is merged to master by olivier, now adding the targettest

**Comment by jonathanraes** [comment]
clean now except for a single checkstyle error and some trouble in testRespawn for Target.
Please review

File: src/main/java/com/funkydonkies/w4v3/App.java [hunk 1]
```diff
@@ -115,7 +100,7 @@ public void simpleInitApp() {
 
 	@Override
 	public void simpleUpdate(final float tpf) {
-		clBox.move(tpf);
+		movBox.move(tpf);
```
Comment (Danilodum) [inline src/main/java/com/funkydonkies/w4v3/App.java:104]: refactoring should be done in a different branch. please keep this in mind next time

File: src/main/java/com/funkydonkies/w4v3/obstacles/MovingBox.java [hunk 2]
```diff
@@ -43,12 +47,11 @@ public MovingBox(final double width, final double height, final double depth, fi
...
 	@Override
 	public final void draw(final Material mat, final PhysicsSpace psySpace) {
-		final float xTrans = 20f;
-		final float zTrans = 0.5f;
-		geom.move(xTrans, 0, zTrans);
 		geom.setMaterial(mat);
 		geom.addControl(phys);
 		psySpace.add(phys);
+		phys.setPhysicsLocation(new Vector3f().add((float) getxCoord(), (float) getyCoord(), (float) getzCoord()));
+		geom.setLocalTranslation(phys.getPhysicsLocation());
```
Comment (Danilodum) [inline src/main/java/com/funkydonkies/w4v3/obstacles/MovingBox.java:54]: is this line necessary?

File: src/main/java/com/funkydonkies/w4v3/obstacles/Obstacle.java [hunk 3]
```diff
@@ -13,6 +13,9 @@
 	private double width;
 	private double height;
 	private double depth;
+	private double xCoord;
+	private double yCoord;
```
Comment (Danilodum) [inline src/main/java/com/funkydonkies/w4v3/obstacles/Obstacle.java]: doubles? why?

File: src/test/java/com/funkydonkies/w4v3/obstacles/TargetTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		final Vector3f startLoc = tar.getLocation();
+		System.out.println(startLoc);
+		tar.respawn();
+		Vector3f newLoc = tar.getLocation();
+		System.out.println(newLoc);
+		System.out.println(startLoc);
+//		assertNotEquals(tar.getLocation(), startLoc);
+	}
+	
+//	/**
```
Comment (Danilodum) [inline src/test/java/com/funkydonkies/w4v3/obstacles/TargetTest.java]: commented code should be removed

File: src/test/java/com/funkydonkies/w4v3/obstacles/ObstacleFactoryTest.java [hunk 5]
```diff
@@ -39,19 +54,27 @@ public static void setUpBeforeClass() {
...
 	
-	/**
-	 * Test if the not valid objects aren't instantiated.
-	 */
 	@Test
-	public final void instantiateNotExisting() {
-		final Obstacle notValidOb = obF.makeObstacle("NOTVALID", 2, 4, 1,
-				node, assetManager, combo);
-		assertEquals(notValidOb, null);
+	public final void testMakeTarget(){
```
Comment (Danilodum14) [inline src/test/java/com/funkydonkies/w4v3/obstacles/ObstacleFactoryTest.java]: test has no javadoc

**Comment by Danilodum** [comment]
octopull is working!

File: src/main/java/com/funkydonkies/w4v3/obstacles/Target.java [hunk 6]
```diff
@@ -21,27 +20,28 @@
...
 	 * @param h the height of the target
 	 * @param d the depth of the target
+	 * @param x the x coordinate of the box
+	 * @param y the y coordinate of the box
+	 * @param z the z coordinate of the box
 	 * @param node Node to attach the target to the scene
-	 * @param assetManager assetmanager to create the material
 	 * @param combo Combo to create target control
 	 */
-	public Target(final double w, final double h, final double d, final Node node,
```
Comment (Danilodum14) [inline src/main/java/com/funkydonkies/w4v3/obstacles/Target.java]: *Auto-generated comment*

    More than 7 parameters (found 8).

File: src/main/java/com/funkydonkies/w4v3/obstacles/Target.java [hunk 7]
```diff
@@ -21,27 +20,28 @@
...
+	 * @param z the z coordinate of the box
 	 * @param node Node to attach the target to the scene
-	 * @param assetManager assetmanager to create the material
 	 * @param combo Combo to create target control
 	 */
-	public Target(final double w, final double h, final double d, final Node node, 
-			final AssetManager assetManager, final Combo combo) {
-		super(w, h, d, node);
-		box = new Box((float) w, (float) h, (float) d);
+	public Target(final float w, final float h, final float d,
```
Comment (Danilodum14) [inline src/main/java/com/funkydonkies/w4v3/obstacles/Target.java]: Is combo necessary to pass?

File: src/test/java/com/funkydonkies/w4v3/obstacles/TargetTest.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+	@Test 
+	public final void testRespawn() {
+		tar.draw(mat, psySpace);
+		final Vector3f startLoc = tar.getLocation();
+		System.out.println(startLoc);
+		tar.respawn();
+		Vector3f newLoc = tar.getLocation();
+		System.out.println(newLoc);
+		System.out.println(startLoc);
+//		assertNotEquals(tar.getLocation(), startLoc);
```
Comment (jonathanraes) [inline src/test/java/com/funkydonkies/w4v3/obstacles/TargetTest.java:86]: This is pretty weird

File: src/main/java/com/funkydonkies/w4v3/obstacles/Obstacle.java [hunk 9]
```diff
@@ -21,14 +24,21 @@
 	 * @param h the height of the obstacle
 	 * @param d the depth of the obstacle
 	 * @param n Node to attach the obstacle to the scene
+	 * @param x the x coordinate of the obstacle
+	 * @param y the y coordinate of the obstacle
+	 * @param z the z coordinate of the obstacle
 	 */
-	public Obstacle(final double w, final double h, final double d, final Node n) {
+	public Obstacle(final float w, final float h, final float d, final float x,
```
Comment (Danilodum14) [inline src/main/java/com/funkydonkies/w4v3/obstacles/Obstacle.java]: much better :+1:

**Comment by Danilodum14** [comment]
No tool errors. Merging.

File: src/test/java/com/funkydonkies/w4v3/obstacles/TargetTest.java (hunk 8 above)
Comment (Danilodum) [inline src/test/java/com/funkydonkies/w4v3/obstacles/TargetTest.java:86]: The target also seems to count several combos at once sometimes. not stable.

**Comment by jonathanraes** [comment]
still waiting to be merged...
Target is already in master and this is needed for the TargetTest and refactorings

**Pull request merged**